smrf/distribute/image_data.py

```diff
@@ -62,6 +62,18 @@ def thread_variables(self):
 
         return self._thread_variables
 
+    def add_thread_variables(self, variables):
+        """Add a list or single variable to the thread variables
+
+        Args:
+            variables (list or str): List or string of variables to add
+        """
+
+        if isinstance(variables, str):
+            variables = list(variables)
+
+        self._thread_variables = self.thread_variables + variables
```

Comment:
I think this will cause an issue, where the method will be a one time use and successive of the method will 'forget' previously added ones.

I think going with a property setter and a separate add method is the way to go:
```python
    @thread_variables.setter
    def thread_variables(self, value):
        self._thread_variables = value

    def add_thread_variable(self, variables):
        if isinstance(variables, str):
            variables = list(variables)
            
        self.thread_variables = self.thread_variables + variables
```

Something like this (untested code)...

Reply:
Makes sense, the setter works